### PR TITLE
unsafe-require/typed #:opaque doesn't warn about opaque structs passed as Any

### PR DESCRIPTION
**Pull request by AlexKnauth**
This changes how the `#:opaque` option in `unsafe-require/typed` behaves. It changes the contract on the predicate to allow opaque structs to be passed as `Any` without displaying a warning.

This fixes an issue raise in https://groups.google.com/forum/#!topic/racket-users/-pgP7-C_3D4, so that it can work if you use `unsafe-require/typed`

### CODE REVIEW

File: typed-racket-lib/typed-racket/base-env/prims-contract.rkt [hunk 1]
```diff
@@ -391,9 +393,13 @@
            #,(if (attribute ne)
                  (internal (syntax/loc stx (define-type-alias-internal ty (Opaque pred))))
                  (syntax/loc stx (define-type-alias ty (Opaque pred))))
-           #,(ignore #'(define pred-cnt
-                         (or/c struct-predicate-procedure?/c
-                               (any-wrap-warning/c . c-> . boolean?))))
+           #,(if (attribute unsafe)
+                 (ignore #'(define pred-cnt
```
Comment (samth) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: Just use `(define pred-cnt any/c)`.

**Comment by samth** [comment]
Let's add a test as well.

File: typed-racket-lib/typed-racket/base-env/prims-contract.rkt (hunk 1 above)
Comment (AlexKnauth) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: Shouldn't it still attempt to check that it is a predicate? I was trying to make the change in behavior as minimal as possible for this

Comment (samth) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: If we wanted that, we can just do `(any/c . ->.  boolean?)`. But I don't think we should add checks for `unsafe-`.

Comment (AlexKnauth) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: I'll change it to use `predicate/c` then

Comment (samth) [inline typed-racket-lib/typed-racket/base-env/prims-contract.rkt]: No, really, let's not use a contract. For example, when testing the performance impact of contracts, it's nice to be able to use `unsafe-require/typed`. Sometimes adding contracts defeats that.

**Comment by AlexKnauth** [comment]
Why not leave the `predicate/c` check there? It doesn't wrap the argument like `(-> any-wrap/c boolean?)` would, so it only allocates one wrapper ever.

**Comment by samth** [comment]
Adding wrappers for unsafe forms is not right. The point of unsafe is to not have checks at all, not just to avoid tricky checks.

**Comment by AlexKnauth** [comment]
Ok, I've changed it to `any/c`. (I don't think I can change it to not use `require/contract`, because that does other stuff)

**Comment by AlexKnauth** [comment]
Should this be merged?

**Pull request merged**